src/modules/0.0.17/aws_secrets_manager/index.ts

```diff
@@ -120,11 +124,12 @@ export const AwsSecretsManagerModule: Module2 = new Module2(
 
                     rawSecret = await getSecret(
                       client.secretsClient,
-                      secretName
+                      secretAWS.Name!
                     );
                     i++;
                   } while (!rawSecret && (i<30));
-                  secret.name = secretName;
+                  secret.name = secretAWS.Name!;
```

Comment:
Why allow the output of `getSecret` to be `undefined` and then assert that it is not to Typescript here? Better to make it return the result without changes and `throw` an error if it's undefined, I think.